test/unit_tests/entity/ConversationSpec.js

```diff
@@ -794,44 +794,115 @@ describe('Conversation', () => {
   });
 
   describe('shouldUnarchive', () => {
-    let time = undefined;
+    let timestamp = undefined;
+
+    const conversationEntity = new z.entity.Conversation(z.util.createRandomUuid());
+
+    const selfUserEntity = new z.entity.User(z.util.createRandomUuid());
+    selfUserEntity.is_me = true;
+    selfUserEntity.inTeam(true);
+    conversationEntity.self = selfUserEntity;
 
     beforeEach(() => {
-      time = Date.now();
-      conversation_et.archivedTimestamp(time);
-      conversation_et.archivedState(true);
-      conversation_et.notificationState(z.conversation.NotificationSetting.STATE.NOTHING);
+      timestamp = Date.now();
+      conversationEntity.archivedTimestamp(timestamp);
+      conversationEntity.archivedState(true);
     });
 
-    it('returns expected bool whether a conversation should be unarchived', () => {
-      conversation_et.last_event_timestamp(time - 100);
-      expect(conversation_et.shouldUnarchive()).toBeFalsy();
+    afterEach(() => conversationEntity.messages_unordered.removeAll());
 
-      conversation_et.last_event_timestamp(time);
-      expect(conversation_et.shouldUnarchive()).toBeFalsy();
+    it('returns false if conversation is not archived', () => {
+      conversationEntity.archivedState(false);
+      expect(conversationEntity.shouldUnarchive()).toBe(false);
+    });
 
-      conversation_et.last_event_timestamp(time + 100);
-      expect(conversation_et.shouldUnarchive()).toBeFalsy();
+    it('returns false if conversation is in no notification state', () => {
+      conversationEntity.notificationState(z.conversation.NotificationSetting.STATE.NOTHING);
```

Comment:
Would be good to still add a message in the conversation (let say the highest priority type of message, namely `mention`), just to check that the test is not successful only because there are no messages in there